lib/std/fmt/parse_float/convert_fast.zig

```diff
@@ -0,0 +1,130 @@
+            1e0, 1e1, 1e2,  1e3, 1e4, 1e5, 1e6, 1e7,
+            1e8, 1e9, 1e10, 0,   0,   0,   0,   0,
+        })[i & 15],
+
+        f64 => ([32]f64{
+            1e0,  1e1,  1e2,  1e3,  1e4,  1e5,  1e6,  1e7,
+            1e8,  1e9,  1e10, 1e11, 1e12, 1e13, 1e14, 1e15,
+            1e16, 1e17, 1e18, 1e19, 1e20, 1e21, 1e22, 0,
+            0,    0,    0,    0,    0,    0,    0,    0,
+        })[i & 31],
+
+        f128 => ([64]f128{
+            1e0,  1e1,  1e2,  1e3,  1e4,  1e5,  1e6,  1e7,
+            1e8,  1e9,  1e10, 1e11, 1e12, 1e13, 1e14, 1e15,
+            1e16, 1e17, 1e18, 1e19, 1e20, 1e21, 1e22, 1e23,
+            1e24, 1e25, 1e26, 1e27, 1e28, 1e29, 1e30, 1e31,
+            1e32, 1e33, 1e34, 1e35, 1e36, 1e37, 1e38, 1e39,
+            1e40, 1e41, 1e42, 1e43, 1e44, 1e45, 1e46, 1e47,
+            1e48, 0,    0,    0,    0,    0,    0,    0,
+            0,    0,    0,    0,    0,    0,    0,    0,
+        })[i & 63],
+
+        else => unreachable,
+    };
+}
+
+fn fastIntPow10(comptime T: type, i: usize) T {
+    return switch (T) {
+        u64 => ([16]u64{
+            1,             10,             100,             1000,
+            10000,         100000,         1000000,         10000000,
+            100000000,     1000000000,     10000000000,     100000000000,
+            1000000000000, 10000000000000, 100000000000000, 1000000000000000,
+        })[i],
+
+        u128 => ([35]u128{
+            1,                                   10,
+            100,                                 1000,
+            10000,                               100000,
+            1000000,                             10000000,
+            100000000,                           1000000000,
+            10000000000,                         100000000000,
+            1000000000000,                       10000000000000,
+            100000000000000,                     1000000000000000,
+            10000000000000000,                   100000000000000000,
+            1000000000000000000,                 10000000000000000000,
+            100000000000000000000,               1000000000000000000000,
+            10000000000000000000000,             100000000000000000000000,
+            1000000000000000000000000,           10000000000000000000000000,
+            100000000000000000000000000,         1000000000000000000000000000,
+            10000000000000000000000000000,       100000000000000000000000000000,
+            1000000000000000000000000000000,     10000000000000000000000000000000,
+            100000000000000000000000000000000,   1000000000000000000000000000000000,
+            10000000000000000000000000000000000,
+        })[i],
+
+        else => unreachable,
+    };
+}
+
+pub fn convertFast(comptime T: type, n: Number(T)) ?T {
+    const MantissaT = common.mantissaType(T);
+
+    if (!isFastPath(T, n)) {
+        return null;
+    }
+
+    // TODO: x86 (no SSE/SSE2) requires x87 FPU to be setup correctly with fldcw
```

Comment:
This is noted in the algorithm's paper. Can see an example implementation in the rust tree: https://github.com/rust-lang/rust/blob/6b6c1ffacc5df738b3560369746d87499adbeae1/library/core/src/num/dec2flt/fpu.rs

Skipped implementation for the moment since I do not have a test setup with this, but should be easy to implement.